stata_kernel/config.py

```diff
@@ -106,10 +109,6 @@ def get(self, key, backup=None):
         return self.env.get(key, backup)
 
     def set(self, key, val, permanent=False):
```

Comment:
Why is this deleted? Wouldn't it be necessary for setting  the `cache_dir` interactively?

Reply:
Yes, I agree. I overlooked support for setting `cache_dir` interactively.

I've consolidated the logic for setting `cache_dir` into the `set` function so that it isn't duplicated in the `__init__` method.

Reply:
There are three issues with this:

- My kernel crashes when I try to set the directory and then run any command. Unsure why.
- Setting the cache directory interactively would now set it to have a temporary directory. If it's set permanently then the temporary directory would persist across sessions.
- I also don't think it's necessary to set a new temporary directory every time it's set interactively in the first place (though I can more or less see why it could be useful).

_However_, I think that your previous change was fine. If you revert and add these lines back in as they were I can merge it. Of course, feel free to try to fix the above issues, but as I said I don't know that it's necessary to do that.

Reply:
Thanks, that makes total sense. Sorry for over-complicating it.

I went ahead and put the code in the `set` method back in as you suggested.